Added section in Quickstart on graphing

rpm4 opened the pull request:
I am bad at programming and figured I should add a blurb on explaining how to import your OWN data files for manipulation. Later on, I intend to expand the "graphing multiple lines" page as well.

## Changes

please describe your changes here :smiley:

## Checklist

- [ ] added tests, if applicable
- [ ] updated documentation, if applicable
- [ ] updated CHANGELOG.md
- [ ] tests pass

DLafayetteII reviewed (changes requested):
I believe the he process you are walking new users through is meant to be covered under 'Quick Start - Create a Data object', and then Quick Start - Visualize Data (i.e. graphing) - quick1D. What I think your version does better is in providing an example to novice users of what a file path looks like. I think it would be a better use of your work to update the existing Quick Start - Create a Data Object and Visualize Data, quick1D sections instead of tacking this all on the end of the existing Quick Start.

In the existing 'Create a Data Object' section, they use wt.open() since their path leads to a wt5 file. In yours, you open data using 'd = wt.data.from_Tensort27(). If you are giving either example opening data ( a from method or an open method), it is probably useful to add a sentence noting that we have this difference between wt5 and non wt5 file types, and pointing new users to http://wright.tools/en/stable/data.html#instantiation for more info like you currently do at the end of this section.

An issue I am noticing is that is outside the scope of your pull is that the section on Data - Instantiation does not mention wt.open(), which was added at a later date.

rpm4 commented:
^ I addressed some of these comments with some new words and I agree that it should be moved around. Any thoughts from others ?

ksunden commented on docs/quickstart.rst:
```diff
@@ -323,6 +323,54 @@ Much like data objects, collection objects have a method :meth:`~WrightTools.col
 
 Collections can be saved inside of wt5 files, so be aware that :meth:`~WrightTools.open` may return a collection or a data object based on the contents of your wt5 file.
 
+Manipulating Your Data -- A 1D Plot Example
+----------------------
+Now that you have learned how to manipulate data provided within WrightTools itself, it is time to investigate manipulation of personal, experimental data. Above, data was already put in WrightTools and was in the .wt5 file format, which permitted use of 'wt.open()'. Here, this will no longer be the case, and other avenues must be explored for performing analysis in WrightTools. Below we will inv ...
```
```suggestion
Manipulating Your Data -- A 1D Plot Example
-------------------------------------------

Now that you have learned how to manipulate data provided within WrightTools itself, it is time to investigate manipulation of personal, experimental data. Above, data was already put in WrightTools and was in the .wt5 file format, which permitted use of 'wt.open()'. Here, this will no longer be the case, and other avenues must be explored for performing analysis in WrightTools. Below we will investigate plotting an infrared spectrum from a typical laboratory instrument (Bruker Tensor 27).
```

ksunden replied:
just some formatting, this file format can be a little persnickety (markdown is better in this regard, but this is the standard for python docs, though that is changing)

ksunden commented on docs/quickstart.rst:
```diff
@@ -323,6 +323,54 @@ Much like data objects, collection objects have a method :meth:`~WrightTools.col
+
+.. code-block:: python
+
+   d = wt.data.from_Tensor27('C:\\Users\\user\\Desktop\\IR_spec.dpt')
```
```suggestion
   >>> d = wt.data.from_Tensor27('C:\\Users\\user\\Desktop\\IR_spec.dpt')
```

Just consistency with adding the python prompt for things typed into a python shell

ksunden commented on docs/quickstart.rst:
```diff
@@ -323,6 +323,54 @@ Much like data objects, collection objects have a method :meth:`~WrightTools.col
+
+.. code-block:: python
+
+   wt.artists.quick1D(d)
```
```suggestion
   >>> wt.artists.quick1D(d)
```

(Same consistency in formatting)

ksunden commented:
I have no substantive objection to this being added (outside of minor formatting consistency)

ksunden commented:
@DLafayetteII are your concerns addressed, if so I think we can merge this

DLafayetteII reviewed (commented):
@ksunden  I still do not approve.  If @rpm4  did not find the quick start helpful as a new user, I approve of him updating the section to be more useful. However, the existing Quick Start section begins by telling you how to open (or create) a data object, and then visualize it, the first example of which is 1D. If the goal of this new section is to provide a more detailed walkthrough of importing data and plotting 1D,  I think we should just update the existing section to be more detailed. Having multiple sections with the same purpose in different places within the quick start section seems confusing.

rpm4 commented:
Ok cool sounds good -- I'll rewrite this tomorrow re: David's requests because I think he's right -- the issue is that this will inevitably be a piecewise subject as I don't have any 2D data sets to easily manipulate atm. However, I can start with the 1D and add on from there.

rpm4 commented:
@ksunden @DLafayetteII Okay I think I addressed most of the issues that David had. Let me know what you think. It is a temporary bandaid, but I frankly need to become more used to WrightTools to further adjust/amend issues with Quickstart. For now, at least, a walk through is provided for graphing a quick 1D plot from personal data.

kameyer226 commented:
@DLafayetteII  for approval as ksunden busy

ddkohler commented:
I also do not approve of this in its current form.  
* I think the amount of discourse is too much.  I think a "quickstart" should be more concise.  We can refer the reader elsewhere if they need a background in how python/file paths work.
* `quickstart.rst` has conflicts that need to be resolved; changes from the last edit have not been incorporated.
* This PR is not running our pytest tests--I suspect this is because we did not configure our triggering events correctly at the time this was submitted.  WIthout running tests, we risk our docs build failing.  I suggest that if we want to move forward with these changes, we update from master so that the updated CI rules trigger.

Keep in mind we can refer the reader to our "examples" folder, or we can also put examples in method docstrings...

@rpm4 let us know if you are still committed to this; I imagine this got abandoned since it's been such a long time...

rpm4 commented:
Yeah this is ancient history, I revamped this last June with you Blaise and Kyle.
________________________________
From: Daniel Kohler ***@***.***>
Sent: Monday, April 1, 2024 10:24 AM
To: wright-group/WrightTools ***@***.***>
Cc: Ryan McDonnell ***@***.***>; Mention ***@***.***>
Subject: Re: [wright-group/WrightTools] Added section in Quickstart on graphing (PR #1087)


I also do not approve of this in its current form.

  *   I think the amount of discourse is too much. I think a "quickstart" should be more concise. We can refer the reader elsewhere if they need a background in how python/file paths work.
  *   quickstart.rst has conflicts that need to be resolved; changes from the last edit have not been incorporated.
  *   This PR is not running our pytest tests--I suspect this is because we did not configure our triggering events correctly at the time this was submitted. WIthout running tests, we risk our docs build failing. I suggest that if we want to move forward with these changes, we update from master so that the updated CI rules trigger.

Keep in mind we can refer the reader to our "examples" folder, or we can also put examples in method docstrings...

@rpm4<https://urldefense.com/v3/__https://github.com/rpm4__;!!Mak6IKo!MKmkpC1c8TXYNHxk9jainX8k7CZOPMeD8Jyy-8yJNh9MhYq4ojDA_dGBBzsDzwajp4mF7NJbxLjQWrLEihpkTiuOfTI$> let us know if you are still committed to this; I imagine this got abandoned since it's been such a long time...

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https://github.com/wright-group/WrightTools/pull/1087*issuecomment-2029957852__;Iw!!Mak6IKo!MKmkpC1c8TXYNHxk9jainX8k7CZOPMeD8Jyy-8yJNh9MhYq4ojDA_dGBBzsDzwajp4mF7NJbxLjQWrLEihpkP1MGmnk$>, or unsubscribe<https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AX2KBGEGMQEE2YLSE2SK4HLY3F33HAVCNFSM54L2Y752U5DIOJSWCZC7NNSXTN2JONZXKZKDN5WW2ZLOOQ5TEMBSHE4TKNZYGUZA__;!!Mak6IKo!MKmkpC1c8TXYNHxk9jainX8k7CZOPMeD8Jyy-8yJNh9MhYq4ojDA_dGBBzsDzwajp4mF7NJbxLjQWrLEihpkv6ToY1k$>.
You are receiving this because you were mentioned.Message ID: ***@***.***>
